overlord/devicestate/devicestate_serial_test.go

```diff
@@ -1882,10 +1882,16 @@ func (s *deviceMgrSerialSuite) TestDeviceRegistrationNotInInstallMode(c *C) {
 	st.Set("seeded", true)
 	// set run mode to "install"
 	devicestate.SetSystemMode(s.mgr, "install")
+
+	// one of the startup managers needs to access the gadget snap to proceed
+	r2 := devicestatetest.MockGadget(c, s.state, "pc", snap.R(2), nil)
```

Comment:
shouldn't we set ensureInstalledRan instead perhaps? or did this test always run install itself?

Reply:
this test was always running install itself, but mocking ensureInstalledRan seems more logical